### PR TITLE
Report builder ko3.0

### PR DESCRIPTION
**Pull request by nickpell**
@NoahCarnahan @orangejenny @benrudolph 

Although report builder is scheduled to be upgraded to bootstrap 3, this will finally fix http://manage.dimagi.com/default.asp?182553 in the meantime

Requires https://github.com/dimagi/commcare-hq/pull/8478

### CODE REVIEW

File: corehq/apps/style/templates/style/bootstrap2/base.html [hunk 1]
```diff
@@ -30,7 +30,12 @@
         {% else %}
         <script src="{% static 'jquery-1.7.1-legacy/jquery.js' %}"></script>
         {% endif %}
+
+        {% if upgrade_knockout_js %}
+        <script type="text/javascript" src="{% new_static 'knockout/dist/knockout.js' %}"></script>
+        {% else %}
         <script type="text/javascript" src="{% static 'knockout-2.3.0-legacy/knockout.js' %}"></script>
```
Comment (NoahCarnahan) [inline corehq/apps/style/templates/style/bootstrap2/base.html]: Should this line be removed?

Comment (benrudolph) [inline corehq/apps/style/templates/style/bootstrap2/base.html]: no, this is the `base.html` so any page without `upgrade_knockout_js` set to True won't have it loaded at all

Comment (NoahCarnahan) [inline corehq/apps/style/templates/style/bootstrap2/base.html]: Oh I'm silly, missed the `if ... else`.

File: corehq/apps/style/decorators.py [hunk 2]
```diff
@@ -52,3 +52,15 @@ def wrapped(request, *args, **kwargs):
             return fn(request, *args, **kwargs)
         return wrapped
     return decorate
+
+
+def upgrade_knockout_js():
```
Comment (benrudolph) [inline corehq/apps/style/decorators.py]: i actually think this is kind of an antipattern: https://github.com/dimagi/commcare-hq/pull/8499/files for an example.

i believe that example allows us not to use `@method_decorator`

**Comment by nickpell** [comment]
build failure real, will fix

**Comment by nickpell** [comment]
opening new branch without merge conflicts

**Pull request closed without merging**